### ISSUE TITLE
Python: Error in notebook 00-getting-started "field "memory" not yet prepared"

### ISSUE THREAD

**Issue by sqlshep** [microsoft/semantic-kernel#3104]
**Describe the bug**
While running python **notebook 00-getting-started**, in the last cell, that performs the "print(joke_function("time travel to dinosaur age"))". I receive the following error

**"ConfigError: field "memory" not yet prepared so type is still a ForwardRef, you might need to call SKContext.update_forward_refs()." **

I tried the SK Code in three notebooks, get the same error in all of them 

**To Reproduce**
Load and run **notebook 00-getting-started** all the cells to the last cell 

**Expected behavior**
Preferably not fail... 

**Screenshots**

**Platform**
 - OS: [Mac]
 - IDE: [Jupyter Notebook]
 - Language: [Python]
 - Source: [cloned main repository]

**Additional context**
None

**Comment by AdarshAcharya5** [comment]
@sqlshep I tried running the notebook and it seems to work fine. (Windows 10)

![Screenshot (489)](https://github.com/microsoft/semantic-kernel/assets/132294330/1491cd98-f994-4de5-8d18-e12fe80caeec)

Could you provide more context?. Also was your kernel using AzureChatCompletion or OpenAIChatCompletion service.?

**Comment by sqlshep** [comment]
Python 3.8.8

It is from this notebook,
https://github.com/microsoft/semantic-kernel/blob/main/python/notebooks/00-getting-started.ipynb 
i made no changes to it or the skill, or anything for that matter. 

I tried GPT-3.5 and GPT-4

**Comment by MikeAinOz** [comment]
Is there a work-around for this problem?

**Comment by MikeAinOz** [comment]
Hi @matthewbolanos , I'm trying to find the status on this issue/bug, can you help me? I know it's out there I just don't know how to find it.

**Comment by MikeAinOz** [comment]
I have a work around @sqlshep @matthewbolanos .  The problem appears to be the pydantic package. I use conda, so i created a new environment, made sure there was no pydantic package, installed ipykernel and now the notebook runs ok

**Comment by matthewbolanos** [comment]
Assigning @moonbox3 to take a look at this issue to see why it might be happening.

**Comment by MikeAinOz** [comment]
The environment that caused this issue is still available so that the error can be reproduced
________________________________

**Comment by moonbox3** [comment]
Hi @MikeAinOz, @sqlshep

if you do a `pip list` can you confirm the semantic-kernel installed version shows as:

`semantic-kernel           0.3.15.dev0`

For my understanding: do you use virtual environments? I'm going to have to find access to an Apple device, potentially, to try and repro.

**Comment by MikeAinOz** [comment]
This is running a in an Anaconda environment under Window 11:
(Stats) C:\Users\mike>python
Python 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:17:03) [MSC v.1929 64 bit (AMD64)] on win32

Package                       Version
----------------------------- -----------
aiofiles                      23.2.1
aiohttp                       3.8.6
aiolimiter                    1.0.0
aiosignal                     1.3.1
alabaster                     0.7.12
ansi2html                     0.0.0
anyio                         3.6.1
argh                          0.26.2
arrow                         1.2.2
asgiref                       3.7.2
astroid                       2.11.2
async-timeout                 4.0.3
atomicwrites                  1.4.0
attrs                         23.1.0
autopep8                      1.6.0
Babel                         2.9.1
backcall                      0.2.0
backports.functools-lru-cache 1.6.4
bcrypt                        3.2.0
beautifulsoup4                4.11.1
binaryornot                   0.4.4
black                         22.3.0
bleach                        5.0.0
blis                          0.7.8
brotlipy                      0.7.0
cairocffi                     1.4.0
CairoSVG                      2.6.0
catalogue                     2.0.6
certifi                       2023.7.22
cffi                          1.15.0
chardet                       4.0.0
charset-normalizer            2.0.12
click                         8.1.2
cloudpickle                   2.0.0
colorama                      0.4.4
cookiecutter                  1.7.3
cryptography                  36.0.2
cssselect2                    0.2.1
cycler                        0.11.0
cymem                         2.0.6
cytoolz                       0.12.0
dash                          2.14.0
dash-bootstrap-components     1.5.0
dask                          2022.12.1
dataclasses                   0.8
debugpy                       1.5.1
decorator                     5.1.1
defusedxml                    0.7.1
diff-match-patch              20200713
dill                          0.3.4
dnspython                     2.4.2
docutils                      0.15.2
drawSvg                       1.9.0
english                       2020.7.0
entrypoints                   0.4
fastjsonschema                2.15.3
flake8                        4.0.1
Flask                         2.1.3
fonttools                     4.32.0
frozenlist                    1.4.0
fsspec                        2022.11.0
h11                           0.12.0
h2                            4.1.0
hpack                         4.0.0
httpcore                      0.15.0
httpx                         0.23.0
hyperframe                    6.0.1
idna                          3.3
imagecodecs                   2021.8.26
imageio                       2.23.0
imagesize                     1.3.0
importlib-metadata            4.11.3
importlib-resources           5.6.0
inflection                    0.5.1
interchange                   2021.0.4
intervaltree                  3.0.2
ipykernel                     6.12.1
ipympl                        0.9.2
ipython                       7.32.0
ipython-genutils              0.2.0
ipywidgets                    8.0.4
isodate                       0.6.1
isort                         5.10.1
itsdangerous                  2.1.2
jedi                          0.18.1
jellyfish                     0.9.0
Jinja2                        3.1.1
jinja2-time                   0.2.0
joblib                        1.1.0
jsonschema                    4.19.2
jsonschema-spec               0.2.4
jsonschema-specifications     2023.7.1
jupyter-client                7.3.4
jupyter-core                  4.9.2
jupyterlab-pygments           0.2.0
jupyterlab-widgets            3.0.5
keyring                       23.4.0
kiwisolver                    1.4.2
langcodes                     3.3.0
lazy-object-proxy             1.7.1
locket                        1.0.0
MarkupSafe                    2.1.1
matplotlib                    3.5.1
matplotlib-inline             0.1.3
mccabe                        0.6.1
mistune                       0.8.4
monotonic                     1.5
more-itertools                10.1.0
motor                         3.3.1
multidict                     6.0.4
munkres                       1.1.4
murmurhash                    1.0.7
mypy-extensions               0.4.3
nbclient                      0.5.13
nbconvert                     6.4.5
nbformat                      5.3.0
neobolt                       1.7.17
neotime                       1.7.4
nest-asyncio                  1.5.5
networkx                      2.8.8
numpy                         1.26.1
numpydoc                      1.2.1
openai                        0.28.1
openapi-core                  0.18.1
openapi-schema-validator      0.6.2
openapi-spec-validator        0.6.0
packaging                     21.3
pandas                        1.4.2
pandocfilters                 1.5.0
paramiko                      2.10.3
parse                         1.19.1
parso                         0.8.3
partd                         1.3.0
pathable                      0.4.3
pathspec                      0.9.0
pathy                         0.6.2
patsy                         0.5.2
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        9.1.0
pip                           22.0.4
platformdirs                  2.5.1
plotly                        5.17.0
pluggy                        1.0.0
poyo                          0.5.0
prance                        23.6.21.0
preshed                       3.0.6
prompt-toolkit                3.0.31
psutil                        5.9.0
ptyprocess                    0.7.0
py2neo                        2021.2.3
pycairo                       1.23.0
pycodestyle                   2.8.0
pycparser                     2.21
pydantic                      1.8.2
pydocstyle                    6.1.1
pyflakes                      2.4.0
pyglmnet                      1.1
Pygments                      2.11.2
pylint                        2.13.5
pyls-spyder                   0.4.0
pymongo                       4.6.0
PyNaCl                        1.5.0
pyOpenSSL                     22.0.0
pyparsing                     3.0.7
PyQt5                         5.12.3
PyQt5_sip                     4.19.18
PyQtChart                     5.12
PyQtWebEngine                 5.12.1
pyrsistent                    0.18.1
PySocks                       1.7.1
python-dateutil               2.8.2
python-dotenv                 1.0.0
python-lsp-black              1.2.0
python-lsp-jsonrpc            1.0.0
python-lsp-server             1.5.0
python-slugify                6.1.1
pytz                          2022.1
PyWavelets                    1.3.0
pywin32                       303
pywin32-ctypes                0.2.0
PyYAML                        6.0
pyzmq                         23.2.0
QDarkStyle                    3.0.3
qstylizer                     0.2.1
QtAwesome                     1.1.1
qtconsole                     5.3.2
QtPy                          2.2.1
referencing                   0.30.2
regex                         2023.10.3
requests                      2.31.0
retrying                      1.3.3
rfc3339-validator             0.1.4
rfc3986                       1.5.0
rope                          1.0.0
rpds-py                       0.12.0
Rtree                         1.0.0
ruamel.yaml                   0.18.5
ruamel.yaml.clib              0.2.8
scikit-image                  0.19.3
scikit-learn                  1.0.2
scipy                         1.8.0
semantic-kernel               0.3.15.dev0
setuptools                    62.0.0
shellingham                   1.4.0
six                           1.16.0
smart-open                    5.2.1
sniffio                       1.2.0
snowballstemmer               2.2.0
sortedcontainers              2.4.0
soupsieve                     2.3.1
spacy                         3.3.1
spacy-legacy                  3.0.9
spacy-loggers                 1.0.2
Sphinx                        4.5.0
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
spyder                        5.3.3
spyder-kernels                2.3.3
srsly                         2.4.3
statsmodels                   0.13.2
TA-Lib                        0.4.19
tenacity                      8.2.3
testpath                      0.6.0
text-unidecode                1.3
textdistance                  4.2.2
thinc                         8.0.17
threadpoolctl                 3.1.0
three-merge                   0.1.1
tifffile                      2021.4.8
tinycss2                      1.1.1
toml                          0.10.2
tomli                         2.0.1
toolz                         0.12.0
tornado                       6.1
tqdm                          4.64.0
traitlets                     5.1.1
typed-ast                     1.5.2
typer                         0.4.1
typing_extensions             4.1.1
ujson                         5.2.0
unicodedata2                  14.0.0
Unidecode                     1.3.4
urllib3                       1.26.9
wasabi                        0.9.1
watchdog                      2.1.7
wcwidth                       0.2.5
webencodings                  0.5.1
Werkzeug                      3.0.0
whatthepatch                  1.0.2
wheel                         0.37.1
widgetsnbextension            4.0.5
win-inet-pton                 1.1.0
wrapt                         1.14.0
yapf                          0.32.0
yarl                          1.9.2
zipp                          3.8.0
________________________________

**Comment by moonbox3** [comment]
Hi @MikeAinOz thank you for the update. Glad to hear you are now unblocked. Let's close this issue for now, and if you hit anything else related to this one, feel free to open it up. Otherwise, you can create a new issue, and we'll take a look.
